pkg/instancegroups/rollingupdate.go

```diff
@@ -236,3 +236,14 @@ func sortGroups(groupMap map[string]*cloudinstances.CloudInstanceGroup) []string
 	sort.Strings(groups)
 	return groups
 }
+
+// exitableError inspects an error to determine if the error is
+// fatal enough that the rolling update cannot continue.
+//
+// For example, if a cluster is unable to be validated by the deadline, then it
+// is unlikely that it will validate on the next instance roll, so an early exit as a
+// warning to the user is more appropriate.
+func exitableError(err error) bool {
+	_, ok := err.(*ValidationTimeoutError)
```

Comment:
Use `errors.Is(err, ValidationTimeoutError)` in case we wrap this error at some point.

Reply:
See ac86445 for the implementation of this.

Since the struct takes in arbitrary strings, comparing out of the box via `errors.Is` would always return `false`. I implemented the `Is` interface with this typecheck, so -- if the `ValidationTimeoutError`s need to be differentiated from each other in the future -- it can be done there.